src/js/components/Tag/Tag.js

```diff
@@ -11,23 +11,27 @@ import { Text } from '../Text';
 
 import { StyledRemoveButton, StyledTagButton } from './StyledTag';
 
-const Tag = forwardRef(({ name, value, onRemove, onClick, ...rest }, ref) => {
+const Tag = forwardRef(
+  ({ name, value, size, onRemove, onClick, ...rest }, ref) => {
   const theme = useContext(ThemeContext) || defaultProps.theme;
 
   const containerProps = {
     ref,
     align: 'center',
     background: theme.tag.background,
     border: theme.tag.border,
-    round: theme.tag.round,
+    round: theme.tag.size?.[size]?.round || theme.tag.round,
     ...rest,
   };
   const contents = (
-    <Box width={{ min: 'min-content' }} pad={theme.tag.pad}>
-      <Text>
-        {name && <Text {...theme.tag.name}>{name}</Text>}
-        {name && value ? theme.tag.separator : ''}
-        {value && <Text {...theme.tag.value}>{value}</Text>}
+    <Box
+      width={{ min: 'min-content' }}
+      pad={theme.tag.size?.[size]?.pad || theme.tag.pad}
+    >
+      <Text size={size}>
+        {name && <Text {...theme.tag.name} size={size}> {name}</Text>}
+        {name && value ? <Text size={size}>{theme.tag.separator}</Text> : ''}
+        {value && <Text {...theme.tag.value} size={size}>{value}</Text>}
```

Comment:
I think omitting `size` here would make it easier for theme creators to have a different size for name or value, if we just kept the size at line 31.

Reply:
I tried this but the name and value didn't inherit it